### PR TITLE
Rust: add MacroPat to isExhaustiveMatch

### PR DESCRIPTION
**Pull request by aibaars**
@hvitved

### CODE REVIEW

**Comment by aibaars** [comment]
@paldepind

File: rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll [hunk 1]
```diff
@@ -139,6 +139,9 @@ private predicate isExhaustiveMatch(Pat pat) {
     // `match` expressions must be exhaustive, so last arm cannot fail
     pat = any(MatchExpr me).getLastArm().getPat()
     or
+    // macro patterns is exhaustive if its expansion is
```
Comment (paldepind) [inline rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll:142]:
```suggestion
    // macro invocations are exhaustive if their expansion is
```

File: rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll [hunk 2]
```diff
@@ -149,6 +152,8 @@ private predicate isExhaustiveMatch(Pat pat) {
     or
     pat = parent.(IdentPat).getPat()
     or
+    pat = parent.(MacroPat).getMacroCall().getExpanded()
```
[the remark this answers is not in the document]
Comment (aibaars) [reply rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll:155]: Line 143 is for cases where a macro expands to a pattern that is known to be exhaustive (eg a wildcard) in that case the macro invocation is an exhaustive pattern. 

Line 155 is for the case where a macro invocation appears in a position that should be exhaustive, such as the last arm in a match. In this case whatever the macro expands to is considered to be exhaustive as well.

[the remark this answers is not in the document]
Comment (paldepind) [reply rust/ql/lib/codeql/rust/controlflow/internal/Completion.qll:155]: Thanks, now I get it 😄

**Review by paldepind — APPROVED** [review]
LGTM 🎉

**Pull request merged**